cocotb/decorators.py

```diff
@@ -490,3 +490,41 @@ def f(*args, **kwargs):
 
     def __call__(self, *args, **kwargs):
         return RunningTest(self._func(*args, **kwargs), self)
+
+
+def make_task(coro):
+    """
+    Turns the given object into a task.
+
+    Performs type checking, converts compatible types into :class:`~cocotb.decorators.RunningTask`,
+    and reports common gotchas.
+
+    .. versionadded:: 1.4
+    """
+
+    if isinstance(coro, RunningTask):
+        return coro
+
+    if inspect.iscoroutine(coro):
+        return RunningTask(coro)
+
+    if isinstance(coro, coroutine):
+        raise TypeError(
+            "Attempt to schedule a coroutine that hasn't started: {}.\n"
+            "Did you forget to add parentheses to the @cocotb.test() "
+            "decorator?"
+            .format(coro)
+        )
+
+    if sys.version_info >= (3, 6) and inspect.isasyncgen(coro):
+        raise TypeError(
+            "{} is an async generator, not a coroutine. "
+            "You likely used the yield keyword instead of await.".format(
+                coro.__qualname__))
+
+    raise TypeError(
+        "Attempt to add a object of type {} to the scheduler, which "
```

Comment:
```suggestion
        "Attempt to add an object of type {} to the scheduler, which "
```

Reply:
These strings need to be changed outright, we are not scheduling here.

Reply:
Could you have another look at them?

Reply:
Done.